aptos-move/aptos-gas/src/aptos_framework.rs

```diff
@@ -8,7 +8,71 @@ crate::natives::define_gas_parameters_for_natives!(GasParameters, "aptos_framewo
+    [.algebra.ark_bls12_381_fr_add, { 8.. => "algebra.ark_bls12_381_fr_add" }, 212 * MUL],
+    [.algebra.ark_bls12_381_fr_deser, { 8.. => "algebra.ark_bls12_381_fr_deser" }, 912 * MUL],
+    [.algebra.ark_bls12_381_fr_div, { 8.. => "algebra.ark_bls12_381_fr_div" }, 59_493 * MUL],
+    [.algebra.ark_bls12_381_fr_eq, { 8.. => "algebra.ark_bls12_381_fr_eq" }, 212 * MUL],
+    [.algebra.ark_bls12_381_fr_from_u64, { 8.. => "algebra.ark_bls12_381_fr_from_u64" }, 507 * MUL],
+    [.algebra.ark_bls12_381_fr_inv, { 8.. => "algebra.ark_bls12_381_fr_inv" }, 58_714 * MUL],
+    [.algebra.ark_bls12_381_fr_mul, { 8.. => "algebra.ark_bls12_381_fr_mul" }, 504 * MUL],
+    [.algebra.ark_bls12_381_fr_neg, { 8.. => "algebra.ark_bls12_381_fr_neg" }, 212 * MUL],
+    [.algebra.ark_bls12_381_fr_one, { 8.. => "algebra.ark_bls12_381_fr_one" }, 207 * MUL],
+    [.algebra.ark_bls12_381_fr_serialize, { 8.. => "algebra.ark_bls12_381_fr_serialize" }, 1_308 * MUL],
+    [.algebra.ark_bls12_381_fr_square, { 8.. => "algebra.ark_bls12_381_fr_square" }, 479 * MUL],
+    [.algebra.ark_bls12_381_fr_sub, { 8.. => "algebra.ark_bls12_381_fr_sub" }, 302 * MUL],
+    [.algebra.ark_bls12_381_fr_zero, { 8.. => "algebra.ark_bls12_381_fr_zero" }, 207 * MUL],
+    [.algebra.ark_bls12_381_g1_affine_deser_comp, { 8.. => "algebra.ark_bls12_381_g1_affine_deser_comp" }, 352 * MUL],
+    [.algebra.ark_bls12_381_g1_affine_deser_uncomp, { 8.. => "algebra.ark_bls12_381_g1_affine_deser_uncomp" }, 715_941 * MUL],
+    [.algebra.ark_bls12_381_g1_affine_serialize_comp, { 8.. => "algebra.ark_bls12_381_g1_affine_serialize_comp" }, 2_076 * MUL],
+    [.algebra.ark_bls12_381_g1_affine_serialize_uncomp, { 8.. => "algebra.ark_bls12_381_g1_affine_serialize_uncomp" }, 2_608 * MUL],
+    [.algebra.ark_bls12_381_g1_proj_add, { 8.. => "algebra.ark_bls12_381_g1_proj_add" }, 10_326 * MUL],
+    [.algebra.ark_bls12_381_g1_proj_double, { 8.. => "algebra.ark_bls12_381_g1_proj_double" }, 5_277 * MUL],
+    [.algebra.ark_bls12_381_g1_proj_eq, { 8.. => "algebra.ark_bls12_381_g1_proj_eq" }, 5_057 * MUL],
+    [.algebra.ark_bls12_381_g1_proj_generator, { 8.. => "algebra.ark_bls12_381_g1_proj_generator" }, 11 * MUL],
+    [.algebra.ark_bls12_381_g1_proj_infinity, { 8.. => "algebra.ark_bls12_381_g1_proj_infinity" }, 11 * MUL],
+    [.algebra.ark_bls12_381_g1_proj_neg, { 8.. => "algebra.ark_bls12_381_g1_proj_neg" }, 11 * MUL],
+    [.algebra.ark_bls12_381_g1_proj_scalar_mul, { 8.. => "algebra.ark_bls12_381_g1_proj_scalar_mul" }, 2_502_933 * MUL],
+    [.algebra.ark_bls12_381_g1_proj_sub, { 8.. => "algebra.ark_bls12_381_g1_proj_sub" }, 10_731 * MUL],
+    [.algebra.ark_bls12_381_g1_proj_to_affine, { 8.. => "algebra.ark_bls12_381_g1_proj_to_affine" }, 120_135 * MUL],
+    [.algebra.ark_bls12_381_g2_affine_deser_comp, { 8.. => "algebra.ark_bls12_381_g2_affine_deser_comp" }, 388 * MUL],
+    [.algebra.ark_bls12_381_g2_affine_deser_uncomp, { 8.. => "algebra.ark_bls12_381_g2_affine_deser_uncomp" }, 1_020_285 * MUL],
```

Comment:
Why is un-compressed deserialization so much pricier?

Reply:
It needs a `scalar_mul` to check subgroup membership.

Reply:
Hm, but compressed deserialization does not?

Reply:
Oops the bench had a bug... I just fixed and updated the parameter.

Reply:
Wait, the number is still large... Again, how come uncompressed deserialization needs a `scalar_mul` but compressed one does not?

Reply:
Oh, the new numbers show the compressed deserialization being expensive too.

Reply:
But BLS12-381 admits much more efficient subgroup checks. I'm surprised arkworks doesn't support those.